cmake/modules/gluecodium/gluecodium/Generate.cmake

```diff
@@ -22,44 +22,113 @@ set(includeguard_gluecodium_Generate ON)
+                                                  # (read below).
+      BUILD_OUTPUT_DIR <path>                     # The path to directory where united source files are
+                                                  # prepared. Java, Swift and Dart source files are
+                                                  # concatenated. For C++ 'umbrella' file is prepared
+                                                  # which includes all other generated C++ source files.
+      DART_LIBRARY_NAME                           # Name of the generated library for Dart.
+      DART_FUNCTION_LOOKUP_ERROR_MESSAGE <text>   # Custom error message for when Dart FFI function lookup
+                                                  # fails.
+      WERROR <waning_name1> ...                   # The list of warnings to treat as errors. Possible values:
+                                                  # DocLinks, DeprecatedAttributes, DartOverloads.
+      )
+
+This function invokes the Gluecodium tool based on a set of of input *.lime
+files with a specific target language generator.
+Additional target `${TARGET}.gluecodium.generate` is created to make it possible to
+generate sources without building whole solution.
+
+Following properties will be set for given `target` based on the corresponding option without
+APIGEN_ prefix:
+ * APIGEN_GENERATOR The generator passed.
+ * APIGEN_OUTPUT_DIR Output dir for the main source set, generators will create subdirectories.
+       The main code set generated here is the generated code which depends on input IDL files.
+ * APIGEN_COMMON_OUTPUT_DIR Output dir for the common source set, generators will create
+       subdirectories. The common source set is independent of input IDL files, only depends on
+       options like JAVA_PACKAGE, CPP_INTERNAL_NAMESPACE etc. and can be shared between multiple
+       targets which use the same settings.
+       The output directories can be nested inside another.
+ * APIGEN_BUILD_OUTPUT_DIR Build artifacts created by custom build steps
+       will end up in this directory. This should not be placed in the source output dir since
+       since the generated file cache will delete these build results otherwise.
+
+
+*Name rules*
+Naming of method names, classes, parameters, fields, etc. can be changed according to
+project's needs. It's especially useful for a public API. Look there for a possible
+parameters and more details:
+https://github.com/heremaps/gluecodium/blob/master/docs/naming_conventions.md
+
+*Modularisation*
+If a solution is separated into few modules (for example few SHARED libraries) which use
+Gluecodium and have common primitives generated by Gluecodium (i.e. classes which objects
+may cross boundaries of modules), then such project should use `modularisation` in Gluecodium.
+Man should choose a module which will contain common source code. This common source code stores
```

Comment:
"Man" looks wrong here. Not sure what you meant. If you need a depersonalized "you", use "one".